public/app/features/alerting/unified/hooks/usePagination.ts

```diff
@@ -1,18 +1,24 @@
-import { useEffect, useState } from 'react';
+import { useCallback, useEffect, useMemo, useState } from 'react';
 
 export function usePagination<T>(items: T[], initialPage: number, itemsPerPage: number) {
   const [page, setPage] = useState(initialPage);
 
-  const numberOfPages = Math.ceil(items.length / itemsPerPage);
+  const numberOfPages = useMemo(() => Math.ceil(items.length / itemsPerPage), [items, itemsPerPage]);
```

Comment:
Maybe they do not :) I added `useMemo` because this hook is rendered by many components (each alert rule) so effectively this gets called tens of times on each page rerender